internal/common/resource.go

```diff
@@ -93,8 +97,45 @@ func createOrUpdate(request *Request, resource controllerutil.Object, isClusterR
 
 func setOwner(request *Request, resource controllerutil.Object, isClusterRes bool) error {
 	if isClusterRes {
-		return libhandler.SetOwnerAnnotations(request.Instance, resource)
+		err := libhandler.SetOwnerAnnotations(request.Instance, resource)
+		if err != nil {
+			return err
+		}
+
+		// Removing ownerReferences for objects prior to creation
+		resource.SetOwnerReferences(nil)
+
+		// Removing ownerReferences for existing objects
+		existingRes := newEmptyResource(resource)
+		key, err := client.ObjectKeyFromObject(resource)
+		if err != nil {
+			return err
+		}
+
+		err = request.Client.Get(request.Context, key, existingRes)
+		if err != nil {
+			if !errors.IsNotFound(err) {
```

Comment:
You could combine those checks into one line `err != nil && !errors.IsNotFound(err)` to remove one level of intending.